WordPress/Classes/Networking/DeleteSiteServiceRemote.swift

```diff
@@ -0,0 +1,69 @@
+import Foundation
+import AFNetworking
+
+/// DeleteSiteServiceRemote handles the REST API call to delete a WordPress.com site.
+
+public class DeleteSiteServiceRemote : ServiceRemoteREST
+{
+     /**
+     Deletes the specified WordPress.com site.
+     
+     - parameter siteID:  The WordPress.com ID of the site.
+     - parameter success: Optional success block with no parameters
+     - parameter failure: Optional failure block with NSError parameter
+     */
+    public func deleteSite(siteID: NSNumber, success: (() -> ())?, failure: (NSError -> ())?) {
```

Comment:
Not sure if we have a style guide for it, but it seems like a common pattern to define void blocks like so:
`success: () -> Void, failure: NSError -> Void` or if they are optional as `success: (() -> Void)?, failure: (NSError -> Void)?`
Maybe update globally just for consistency?